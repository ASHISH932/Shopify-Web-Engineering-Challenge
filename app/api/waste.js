const uri = 'https://api.github.com';

export const searchApi = item => fetch(`/api/wastes?keyword=${item}`)
    .then(response => response.json())
    .then(response => ({ response }))
    .catch(error => ({ error }));