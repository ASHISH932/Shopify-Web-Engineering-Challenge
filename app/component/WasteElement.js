import React from 'react';
import injectSheet from 'react-jss';
import ReactHtmlParser from 'react-html-parser';
import PropTypes from 'prop-types';

// export const DataListContainer = (props) => {
const parseHTML = (body) => {
    body = ReactHtmlParser(body) + "";
    if (!body.includes('<li>') || !body.includes('<ul>')) {
        body = `<ul><li>${body}</li></ul>`
    }
    console.log(body)
    return body;
}
export const WasteElement = ({ classes, waste, onClick }) => (
    <div className={classes.main}>
        <p className={waste.favourite ? classes.favourite : classes.normal}>
            <i className="fa fa-star" aria-hidden="true" onClick={e => onClick(e)}></i>
            {waste.title}
        </p>
        <div
        className={classes.list}
        dangerouslySetInnerHTML={{__html: parseHTML(waste.body) }}
        />
    </div>);
    
const styles = {
    main: {
        display: 'grid',
        gridTemplateColumns: '1fr 2fr'
    },
    favourite: {
        '& i':{
            color: '#229261',
            paddingRight: '6px'
        }
    },
    normal: {
        '& i':{
            color: '#aaa',
            paddingRight: '6px'
        }
    },
    list: {
        '& li': {
            paddingBottom: '10px'
        }
    }
}

WasteElement.propTypes = {
    classes: PropTypes.object.isRequired,
    waste: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default injectSheet(styles)(WasteElement);