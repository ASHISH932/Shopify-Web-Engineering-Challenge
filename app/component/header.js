import React from 'react';
import injectSheet from 'react-jss';
import PropTypes from 'prop-types';

export const Header = (props) => {
    const { classes, children } = props;
    return (
        <div className={classes.header}>
            <h1>{children}</h1>
        </div>
    );
};

const styles = {
    header: {
        height: '125px',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        background: 'linear-gradient(to right, #1d5c91 0%, #229261 100%);',
        color: 'white',
        marginBottom: '30px'
    }
}

Header.propTypes = {
    classes: PropTypes.object.isRequired,
    children: PropTypes.object,
};

export default injectSheet(styles)(Header);