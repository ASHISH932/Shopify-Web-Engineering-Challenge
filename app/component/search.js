import React, { Component } from 'react';
import injectSheet from 'react-jss';
import PropTypes from 'prop-types';

export class Search extends Component {
    state = {
      input: '',
    };
    handleClick = (e) => {
        e.preventDefault();
        const val = this.state.input;
        if (val.length > 0) {
            this.props.onSearch(val);
        }
    };
    handleChange = (e) => {
        const a = e.target.value;
        if(a.length > 0) {
            this.setState({
                input: a
            })
        } else {
            this.props.onInputCleared();
        }
    };
    handleKeyDown = (e) => {
        if (e.keyCode === 13 && !e.shiftKey) {
            this.handleClick(e);
        }
    };
    render() {
        const { classes } = this.props;
        return (
            <div className={classes.main}>
                <input type="text" className={classes.text} placeholder="Search" onKeyDown={this.handleKeyDown} onChange={this.handleChange}/>
                <button id="searchButton" onClick={this.handleClick} className={classes.button} aria-pressed="true">
                    <i className="fa fa-search" aria-hidden="true"></i>
                </button>
            </div>
        );
    }
}

const styles = {
    main: {
        margin: 'auto',
        display: 'grid',
        gridTemplateColumns: '15fr 1fr',
        gridGap: '10px',
        padding: '0 2%'
    },
    text: {
        height: 49,
        margin: '4px 2px',
        padding: '8px',
        boxSizing: 'border-box',
        borderRadius: '4px',
        fontSize: '16px'
    },
    button: {
        color: 'white',
        borderRadius: '4px',
        fontSize: '25px',
        margin: '4px 2px',
        cursor: 'pointer',
        backgroundColor:  '#229261 !important',
        border: '1px solid  #229261',
    },
    "search:before" :{
        content: "\f00e"
    },
    "fa-search" : {
        fontSize: '48px'
    }
};


Search.propTypes = {
    classes: PropTypes.object.isRequired,
    onSearch: PropTypes.func.isRequired,
    onInputCleared: PropTypes.func.isRequired,
};

export default injectSheet(styles)(Search);