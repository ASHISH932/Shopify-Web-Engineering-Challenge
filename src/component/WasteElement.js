import React from 'react';
import injectSheet from 'react-jss';

// export const DataListContainer = (props) => {
export const WasteElement = ({ classes, waste, onClick }) => (
    <div className={classes.main}>
        <p className={waste.favourite ? classes.favourite : classes.normal}>
            <i className="fa fa-star" aria-hidden="true" onClick={e => onClick(e)}></i>
            {waste.title}
        </p>
        <div className={classes.list}  dangerouslySetInnerHTML={{__html: waste.body }} />
    </div>);

const styles = {
    main: {
        display: 'grid',
        gridTemplateColumns: '1fr 1fr'
    },
    favourite: {
        '& i':{
            color: '#229261',
            paddingRight: '6px'
        }
    },
    normal: {
        '& i':{
            color: '#aaa',
            paddingRight: '6px'
        }
    },
    list: {
        '& li': {
            paddingBottom: '10px'
        }
    }
}

export default injectSheet(styles)(WasteElement);
// {!!props.header ? <h3>{props.header}</h3> : false}