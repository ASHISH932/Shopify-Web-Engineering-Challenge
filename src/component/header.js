import React from 'react';
import injectSheet from 'react-jss';

export const Header = (props) => {
    const { classes, children } = props;
    return (
        <div className={classes.header}>
            <h1>{children}</h1>
        </div>
    );
};

const styles = {
    header: {
        width: '100%',
        height: '15%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        background: 'linear-gradient(to right, #1d5c91 0%, #229261 100%);',
        color: 'white',
        marginBottom: '30px'
    }
}

export default injectSheet(styles)(Header);