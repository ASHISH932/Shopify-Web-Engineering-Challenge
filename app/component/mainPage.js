import React, { Component } from 'react';
import injectSheet from 'react-jss';
import { compose } from 'recompose';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// Components
import Header from './header';
import Search from './search';
import WasteElement from './WasteElement'

// Selectors and actions
import { startSearch, updateSearchList } from "../actions/searchList";
import { addToFavourites, removeFavourite } from "../actions/favourites";
import searchListSelector from '../selectors/searchListSelector';
import SnackBar from './snackbar';

export class MainPage extends Component {
    
    handleFavouriteClick(waste) {
        waste.favourite ? this.props.removeFavourite(waste.id) : this.props.addToFavourites(waste);
    }

    render() {
        const { classes, favourites, startSearch, updateSearchList, searchList, loadState } = this.props;
        return (
            <div>
                <SnackBar show={loadState} timer={6000}>
                  <p>Loading...</p>
                </SnackBar>
                <Header>
                    Toronto Waste Lookup
                </Header>
                <Search
                onSearch={item => startSearch(item)}
                onInputCleared={() => updateSearchList()}
                />
                <div className = {classes.waste}>
                    {searchList.map(w => <WasteElement key={w.id} waste={w} onClick={() => this.handleFavouriteClick(w)}/>)}
                </div>
                {
                    favourites.length !== 0 ? 
                    <div className = {classes.favourite} >
                        <h2 className={classes.textHeader}> Favourite </h2>
                        {favourites.map(w => <WasteElement key={w.id} waste={w} onClick={() => this.handleFavouriteClick(w)}/>)}
                    </div>
                    :
                    false
                }
            </div>
        );
    }
};

const styles = {
    waste: {
        padding: '0 2%'
    },
    favourite: {
        padding: '0 2%',
        background: '#f7fefa'
    },
    textHeader: {
        fontWeight: 'bold',
        color: '#41a570',
        paddingTop: '20px'
    }
};


const mapStateToProps = ({ searchList, favourites, loadState }) => ({
    favourites,
    searchList: searchListSelector(searchList, favourites),
    loadState,
});

const mapDispatchToProps = dispatch => ({
    startSearch: item => dispatch(startSearch(item)),
    updateSearchList: () => dispatch(updateSearchList([])),
    addToFavourites: waste => dispatch(addToFavourites(waste)),
    removeFavourite: id => dispatch(removeFavourite(id))
});

MainPage.propTypes = {
    classes: PropTypes.object.isRequired,
    favourites: PropTypes.array,
    searchList: PropTypes.array,
    startSearch: PropTypes.func.isRequired,
    updateSearchList: PropTypes.func.isRequired,
    addToFavourites: PropTypes.func.isRequired,
    removeFavourite: PropTypes.func.isRequired,
};

export default compose(
    connect(mapStateToProps, mapDispatchToProps),
    injectSheet(styles),
)(MainPage);