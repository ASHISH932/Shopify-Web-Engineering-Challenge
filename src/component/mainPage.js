import React, { Component } from 'react';
import injectSheet from 'react-jss';
import Header from './header';
import Search from './search';
import WasteElement from './WasteElement'

const waste = {
    body: "<ul><li>Place item in the <strong>Garbage Bin.</strong></li><li>Place item in the <strong>Garbage Bin.</strong></li></ul>",
    title: "Garbage (wrapping and typing)",
    favourite: true, 
}
const wasteList = [];

export const MainPage = (props) => {
    const { classes } = props;
    return (
        <div>
            <Header>
                Toronto Waste Lookup
            </Header>
            <Search />
            <div className = {classes.waste}>
                <WasteElement waste={waste} />
                <WasteElement waste={waste} />
                <WasteElement waste={waste} />
                <WasteElement waste={waste} />
            </div>
            <div className = {classes.favourite} >
                <h3 className={classes.textHeader}> Favourite </h3>
                <WasteElement waste={waste} />
            </div>
        </div>
    );
};


const styles = {
    waste: {
        padding: '0 2%'
    },
    favourite: {
        padding: '0 2%',
        background: '#f7fefa'
    },
    textHeader: {
        fontWeight: 'bold',
        color: '#41a570'
    },
    header: {
        width: '100%',
        height: '15%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        background: 'linear-gradient(to right, #1d5c91 0%, #229261 100%);',
        color: 'white'
    },
    main: {
        display: 'grid',
        gridTemplateColumns: '1fr 1fr',
        height: '85%',
        lineHeight: '25px',
        fontFamily: 'Helvetica, Arial, Lucida Grande, sans-serif',
    },
    list: {
        backgroundColor: '#EEE',
    },
    center: {
        width: '95%',
        marginTop: '5%',
        marginLeft: '5%',
        marginRight: '0%',
    },
    gap: {
        height: '5%',
    }
};

export default injectSheet(styles)(MainPage);