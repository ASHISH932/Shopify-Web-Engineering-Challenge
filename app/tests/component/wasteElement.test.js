import React from 'react';
import { shallow } from 'enzyme';
import { WasteElement } from '../../component/WasteElement';
import waste from '../fixtures/waste';

let wrapper;
let onClickSpy;
beforeEach(() => {
  onClickSpy = jest.fn();

  wrapper = shallow(<WasteElement
    classes={{}}
    waste={waste[0]}
    onClick={onClickSpy}
    />);
});

test('should render WasteElement component correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle favourite click', () => {
    const value = "hey";
    wrapper.find('i').simulate('click', { value });
    expect(onClickSpy).toHaveBeenCalledWith({ value });
  });