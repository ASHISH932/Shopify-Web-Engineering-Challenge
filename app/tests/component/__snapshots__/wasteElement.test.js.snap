// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WasteElement component correctly 1`] = `
<div>
  <p>
    <i
      aria-hidden="true"
      className="fa fa-star"
      onClick={[Function]}
    />
    Garbage (wrapping and typing)
  </p>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": Array [
          <ul>
            <li>
              Place item in the 
              <strong>
                Garbage Bin.
              </strong>
            </li>
            <li>
              Place item in the 
              <strong>
                Garbage Bin.
              </strong>
            </li>
          </ul>,
        ],
      }
    }
  />
</div>
`;
